common/flatpak-dir.c

```diff
@@ -1839,6 +1841,154 @@ get_system_locations (GCancellable *cancellable,
   return g_steal_pointer (&locations);
 }
 
+char **
+flatpak_get_preinstall_config_file_paths (GCancellable *cancellable,
+                                          GError      **error)
+{
+  g_autoptr(GPtrArray) paths = NULL;
+  g_autoptr(GFile) conf_dir = NULL;
+  g_autoptr(GFileEnumerator) dir_enum = NULL;
+  g_autoptr(GError) my_error = NULL;
+  g_autofree char *config_dir = NULL;
+
+  paths = g_ptr_array_new_with_free_func (g_free);
+  config_dir = g_strdup_printf ("%s/%s",
+                                get_config_dir_location (),
+                                SYSCONF_PREINSTALL_DIR);
+
+  if (!g_file_test (config_dir, G_FILE_TEST_IS_DIR))
+    {
+      g_info ("Skipping missing preinstall config directory %s", config_dir);
+      goto out;
+    }
+
+  conf_dir = g_file_new_for_path (config_dir);
+  dir_enum = g_file_enumerate_children (conf_dir,
+                                        G_FILE_ATTRIBUTE_STANDARD_NAME "," G_FILE_ATTRIBUTE_STANDARD_TYPE,
+                                        G_FILE_QUERY_INFO_NONE,
+                                        cancellable, &my_error);
+  if (my_error != NULL)
```

Comment:
```suggestion
  if (!dir_enum)
```

Reply:
Why is that better?

Reply:
It's just a convention that functions return NULL/FALSE when an error occurred, so they work without having to pass the error argument. That makes the error parameter just some additional metadata and the NULL/FALSE acts as the primary mechanism. Also just for consistency.

Reply:
IMO, both are legitimate (and both ways are used in flatpak-dir.c) - sometimes you *need* to check error. I'd probably check dir_enum, but don't think it needs to be changed.